inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,366 @@
+            catch_weight_at_age.resize(nyears*nages*nfleets);
+            unfished_numbers_at_age.resize((nyears+1)*nages);
+            numbers_at_age.resize((nyears+1)*nages);
+            expected_catch.resize(nyears*nfleets);
+            expected_index.resize(nyears*nfleets);
+            biomass.resize((nyears+1));
+            unfished_spawning_biomass.resize((nyears+1));
+            spawning_biomass.resize((nyears+1));
+            log_naa.resize(nages);
+            log_Fmort.resize(nfleets*nyears);
+            log_M.resize(nyears*nages);
+            log_q.resize(nfleets);
+            naa.resize(nages);
+            Fmort.resize(nfleets*nyears);
+            M.resize(nyears*nages);
+            q.resize(nfleets);
+
+            
+            
+        }
+
+        //gets called at each model iteration (used to zero out derived quantities, values summed with +=, etc.)
+        void Prepare() {
+          std::fill(unfished_spawning_biomass.begin(), unfished_spawning_biomass.end(), 0);
+          std::fill(spawning_biomass.begin(), spawning_biomass.end(), 0);
+          std::fill(mortality_F.begin(), mortality_F.end(), 0);
+          std::fill(expected_catch.begin(), expected_catch.end(), 0);
+
+          //Transformation Section
+          for (int age = 0; age <= this->nages; age++) {
+            this -> naa[age] = exp(this -> log_naa[age]);
+            for(int year = 0; year < this->nyears; year++) {
+              int index_ya = year*nages + age;
+              this -> M[index_ya] = exp(this -> log_M[index_ya]);
+            }
+          }
+
+          for(int fleet_ = 0; fleet_ <= this->nfleets; fleet_++) {
+            this -> Fmort[fleet_] = exp(this -> log_Fmort[fleet_]);
+            for(int year = 0; year < this->nyears; year++) {
+              int index_yf = year * nfleets + fleet_;
+              this -> q[index_yf] = exp(this -> log_q[index_yf]);
+            }
+          }
+          // call functions to set up recruitment deviations.
+          this -> recruitment -> PrepareConstrainedDeviations();
+          this -> recruitment.PrepareBiasAdjustment();
+        }
+
+        /**
+         * life history calculations
+         */
+        //delete std::vector<Type> because already declared; function has access to log_naa
+        //exp(this ->log_naa[a]) on line 130 - more explicit that we are reference member
+        inline void CalculateInitialNumbersAA(int index, int a) { //inline all function unless complicated
+          this -> numbers_at_ages[index] = naa[a];  
+        }
+        
+        /**
+        * Calculates mortality at an index, year, and age
+        */
+        void CalculateMortality(int index_ya, int year, int age) {
+          for (int fleet_=0; fleet_ < this -> nfleets; fleet_++) {
+            int index_yf = year*nfleets + fleet_; //index by fleet and years to dimension fold
+            this -> mortality_F[index_ya] += Fmort[index_yf]* this -> fleets[fleet_].selectivity->evaluate(age);
```

Comment:
this -> mortality_F[index_ya] += fleet->Fmort[index_yf]* this -> fleets[fleet_]->selectivity->evaluate(age);

Reply:
So this I think is the reason that some vectors are being mirrored in population and fleet so that we don't have to call out the fleet-> reference every time. Again, we discussed this when developing and weren't clear on the tradeoffs in speed and/or possible errors between the two approaches.